src/CloudStructures/RedisConnection.cs

```diff
@@ -92,38 +118,117 @@ public RedisConnection(RedisConfig config, IValueConverter? converter = null, IC
             {
-                try
+                //--- create inner connection
+                var stopwatch = Stopwatch.StartNew();
+                connection = ConnectionMultiplexer.Connect(this.Config.Options, this.Logger);
+                stopwatch.Stop();
+
+                if (this.Handler is { } handler)
                 {
-                    //--- create inner connection
-                    var stopwatch = Stopwatch.StartNew();
-                    this._connection = ConnectionMultiplexer.Connect(this.Config.Options, this.Logger);
-                    stopwatch.Stop();
-                    this.Handler?.OnConnectionOpened(this, new(stopwatch.Elapsed));
+                    handler.OnConnectionOpened(this, new(stopwatch.Elapsed));
 
                     //--- attach events
-                    this._connection.ConfigurationChanged += (_, e) => this.Handler?.OnConfigurationChanged(this, e);
-                    this._connection.ConfigurationChangedBroadcast += (_, e) => this.Handler?.OnConfigurationChangedBroadcast(this, e);
-                    this._connection.ConnectionFailed += (_, e) => this.Handler?.OnConnectionFailed(this, e);
-                    this._connection.ConnectionRestored += (_, e) => this.Handler?.OnConnectionRestored(this, e);
-                    this._connection.ErrorMessage += (_, e) => this.Handler?.OnErrorMessage(this, e);
-                    this._connection.HashSlotMoved += (_, e) => this.Handler?.OnHashSlotMoved(this, e);
-                    this._connection.InternalError += (_, e) => this.Handler?.OnInternalError(this, e);
-                    this._connection.ServerMaintenanceEvent += (_, e) => this.Handler?.OnServerMaintenanceEvent(this, e);
-                }
-                catch
-                {
-                    this._connection = null;
-                    throw;
+                    connection.ConfigurationChanged += this.OnConfigurationChanged;
+                    connection.ConfigurationChangedBroadcast += this.OnConfigurationChangedBroadcast;
+                    connection.ConnectionFailed += this.OnConnectionFailed;
+                    connection.ConnectionRestored += this.OnConnectionRestored;
+                    connection.ErrorMessage += this.OnErrorMessage;
+                    connection.HashSlotMoved += this.OnHashSlotMoved;
+                    connection.InternalError += this.OnInternalError;
+                    connection.ServerMaintenanceEvent += this.OnServerMaintenanceEvent;
                 }
+
+                return connection;
+            }
+            catch
+            {
+                connection?.Dispose();
+                throw;
             }
-            return this._connection;
         }
     }
+
     private readonly object _gate = new();
-    private ConnectionMultiplexer? _connection = null;
+    private ConnectionMultiplexer? _connection;
     #endregion
+
+    [EditorBrowsable(EditorBrowsableState.Advanced)]
+    public void ReleaseConnection()
```

Comment:
Please provide XML documentation for public accessibility. Only the Summary tag needs to be populated.

Reply:
I added it.